lnclient/lnd/lnd.go

```diff
@@ -448,14 +415,97 @@ func NewLNDService(ctx context.Context, lndAddress, lndCertHex, lndMacaroonHex s
 		return nil, err
 	}
 
-	lndService := &LNDService{client: lndClient, pubkey: info.IdentityPubkey}
+	lndCtx, cancel := context.WithCancel(ctx)
+
+	lndService := &LNDService{client: lndClient, pubkey: info.IdentityPubkey, cancel: cancel}
+
+	// Subscribe to payments
+	go func() {
+		for {
+			paymentStream, err := lndClient.SubscribePayments(lndCtx, &routerrpc.TrackPaymentsRequest{
+				NoInflightUpdates: true,
+			})
+			if err != nil {
+				logger.Logger.Errorf("Error subscribing to payments: %v", err)
+				continue
+			}
+			for {
+				select {
+				case <-lndCtx.Done():
+					return
+				default:
+					payment, err := paymentStream.Recv()
+					if err != nil {
+						logger.Logger.Errorf("Failed to receive payment: %v", err)
+						time.Sleep(2 * time.Second)
+						continue
+					}
+					if payment.Status != lnrpc.Payment_SUCCEEDED {
+						continue
+					}
+
+					logger.Logger.WithFields(logrus.Fields{
+						"payment": payment,
+					}).Info("Received new payment")
+
+					transaction, err := lndPaymentToTransaction(payment)
+					if err != nil {
+						continue
+					}
+
+					eventPublisher.Publish(&events.Event{
+						Event:      "nwc_payment_sent",
+						Properties: transaction,
+					})
+				}
+			}
+		}
+	}()
+
+	// Subscribe to invoices
+	go func() {
+		for {
+			invoiceStream, err := lndClient.SubscribeInvoices(lndCtx, &lnrpc.InvoiceSubscription{})
+			if err != nil {
+				logger.Logger.Errorf("Error subscribing to invoices: %v", err)
```

Comment:
can you please use the JSON logging format - Logger.WithError(err).Error("message")

Reply:
Done!